lua/neotest-kotlin/output_parser.lua

```diff
@@ -146,14 +138,45 @@ end
 ---@param path string
 ---@return table<string, neotest.Result>
 function M.parse_lines(lines, path)
+	---@type neotest.Result[]
 	local results = {}
 	local classes = M.determine_all_classes(path)
 
-	for _, line in ipairs(lines) do
-		local result = M.parse_line(line, classes)
-		if result ~= nil then
-			results[result.id] = result
+	for i, line in ipairs(lines) do
+		if not M.is_valid_gradle_test_line(line, classes) then
+			goto continue
 		end
+
+		local id = M.parse_test_id(line, classes)
+		if not id then
+			goto continue
+		end
+
+		---@type string[]
+		local output = {line}
+		---@type neotest.Error[]
+		local errors = {}
+
+		for j = i+1, #lines do
+			if vim.trim(lines[j]) == "" or M.is_valid_gradle_test_line(lines[j], classes) then
+				break
+			end
+
+			table.insert(errors, { message = lines[j], line = j })
+			table.insert(output, lines[j])
+		end
```

Comment:
This is rather simplistic, but was the most obvious way to do this. Just parse from the current line till the next test case and call that the stacktrace